src/Model/ParticleTracking/prt-prp.f90

```diff
@@ -413,47 +418,52 @@ subroutine release(this, ip, trelease)
     integer(I4B), intent(in) :: ip !< particle index
     real(DP), intent(in) :: trelease !< release time
     ! local
-    integer(I4B) :: irow, icol, ilay, icpl
-    integer(I4B) :: ic, icu
     integer(I4B) :: np
-    real(DP) :: x, y, z
-    real(DP) :: top, bot, hds
     type(ParticleType), pointer :: particle
 
+    call this%initialize_particle(particle, ip, trelease)
+
     ! Increment cumulative particle count
     np = this%nparticles + 1
     this%nparticles = np
 
-    ! Get reduced and user node numbers
-    ic = this%rptnode(ip)
-    icu = this%dis%get_nodeuser(ic)
+    ! Save the particle to the store
+    call this%particles%save_particle(particle, np)
+    deallocate (particle)
 
-    ! Load coordinates and transform if needed
-    x = this%rptx(ip)
-    y = this%rpty(ip)
-    if (this%ilocalz > 0) then
-      top = this%fmi%dis%top(ic)
-      bot = this%fmi%dis%bot(ic)
-      hds = this%fmi%gwfhead(ic)
-      z = bot + this%rptz(ip) * (hds - bot)
-    else
-      z = this%rptz(ip)
-    end if
+    ! Accumulate mass for release point
+    this%rptm(ip) = this%rptm(ip) + DONE
 
-    ! Make sure release point is in the grid/cell
-    call this%validate_release_point(ic, x, y, z)
+  end subroutine release
+
+  subroutine initialize_particle(this, particle, ip, trelease)
```

Comment:
factored out to make the `release()` routine clearer

Reply:
cool